irodsConnector/meta.py

```diff
@@ -28,9 +28,9 @@ def add(self, items: list, key: str, value: str, units: str = None):
             try:
                 item.metadata.add(key.upper(), value, units)
             except irods.exception.CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME:
-                logging.error(kw.RED+"INFO ADD META: Metadata already present"+kw.DEFAULT)
+                logging.error('%sADD META: Metadata already present%s', kw.RED, kw.DEFAULT)
```

Comment:
I think we should remove the colour codes (or at leat move somehow them to the logging handler).

Reply:
As far as I can tell, there is no per-type formatting, only one format per handler.  I think the best I can do is create a set of wrapper functions (or even a single function) to call the logger methods with the appropriate ASCII color codes surrounding the logging message.  Perhaps something like:

```python
def log(log_method, log_message):
    mdict = {
        logging.debug: '%s',
        logging.info: 'kw.YELLOW%skw.DEFAULT',
        logging.warning: 'kw.ORANGE%skw.DEFAULT',
        logging.error: 'kw.RED%skw.DEFAULT',
    }
    log_method(mdict[log_method], log_message)
```

Unfortunately, every single logging statement will have to be updated and the advantage of optional evaluation will be lost.

Reply:
I think a simple string converting function is the best solution.  A little more work when writing the logging statement, but retains all the other advantages.

Reply:
Found the `correct` solution: **add custom formatting targets to the record factory.**

Reply:
See: https://github.com/chStaiger/iBridges-Gui/pull/97/commits/0098e934972b4916f2d29260b69714ca2c6d2a55

Reply:
Looks good! Is it possible to put the `init_logger()` somewhere more central, maybe in utils, so `iBridgesCli.py` can use it too?